src/main/java/org/apache/maven/plugins/deploy/AbstractDeployMojo.java

```diff
@@ -19,16 +19,28 @@
  * under the License.
  */
 
+import java.util.List;
+
+import org.apache.maven.RepositoryUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.metadata.ArtifactMetadata;
 import org.apache.maven.artifact.repository.ArtifactRepository;
 import org.apache.maven.artifact.repository.ArtifactRepositoryPolicy;
 import org.apache.maven.artifact.repository.MavenArtifactRepository;
 import org.apache.maven.artifact.repository.layout.DefaultRepositoryLayout;
 import org.apache.maven.execution.MavenSession;
 import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.Component;
 import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.artifact.ProjectArtifactMetadata;
 import org.apache.maven.rtinfo.RuntimeInformation;
+import org.eclipse.aether.RepositorySystem;
```

Comment:
and soon we will change to org.apache.maven package... ?
maybe think in the other way… why we have not yet change package from maven-resolver?

Reply:
Unsure what you talk about:
* if about package change for resolver, yes, it will happen, somewhere around Maven 5, but that will not affect you as
* Maven 4 introduces new API and _seals Maven internals off_ from plugins, hence from 4.1 or so, plugin will not "tamper" with resolver and other internal things of Maven anymore, will have to go for Maven API
* in short, resolver package change, if that's you refer to will NOT affect you, and if the "soon we will change to" refers to Maven API 4.x, is still WIP and of course non existent in Maven 3.x line.